eras/conway/impl/src/Cardano/Ledger/Conway/Rules/NewEpoch.hs

```diff
@@ -214,9 +200,7 @@ updateRewards ::
   EpochNo ->
   RewardUpdate ->
   Rule (ConwayNEWEPOCH era) 'Transition (EpochState era)
-updateRewards es e ru'@(RewardUpdate dt dr rs_ df _) = do
-  let totRs = sumRewards (es ^. prevPParamsEpochStateL . ppProtocolVersionL) rs_
-  Val.isZero (dt <> dr <> toDeltaCoin totRs <> df) ?! CorruptRewardUpdate ru'
```

Comment:
@lehins this is the location where it was checked, just FYI.

Reply:
Seems like a nice invariant to check, so could you please convert to an STS assertion instead

Reply:
You didn't understand my comment. We want to turn it into an STS Assertion, not a GHC `assert`!

See this example on how this is done: https://github.com/IntersectMBO/cardano-ledger/blob/d90eb4df4651970972d860e95f1a3697a3de8977/eras/shelley/impl/src/Cardano/Ledger/Shelley/Rules/PoolReap.hs#L113-L127

Reply:
Oh got it.

Reply:
After some discussion on the possibilities of doing this, we have opened another [issue](https://github.com/IntersectMBO/cardano-ledger/issues/5036) to track these changes. The current changes are an interim solution.